src/frontends/onnx/frontend/src/op/stft.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/stft.hpp"
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "onnx_import/core/null_node.hpp"
+#include "utils/common.hpp"
+#include "utils/dft.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_17 {
+
+OutputVector stft(const Node& node) {
+    const OutputVector ng_inputs{node.get_ng_inputs()};
+    auto signal = ng_inputs.at(0);
+    const auto dft_length_provided = ng_inputs.size() > 3 && !ngraph::op::is_null(ng_inputs[3]);
+    const auto onesided = node.get_attribute_value<int64_t>("onesided", 1);
+    const int64_t axis = 1;
+
+    const auto& frame_step_node = ng_inputs.at(1);
+    CHECK_VALID_NODE(node,
+                     ngraph::op::is_constant(frame_step_node.get_node_shared_ptr()) &&
+                         ov::shape_size(frame_step_node.get_shape()) <= 1,
+                     "frame_step input must be a scalar or Shape{1} constant.");
+    const auto frame_step =
+        ov::as_type_ptr<default_opset::Constant>(frame_step_node.get_node_shared_ptr())->cast_vector<int64_t>()[0];
+    const auto signal_param_shape = signal.get_partial_shape();
+    CHECK_VALID_NODE(node,
+                     signal_param_shape.rank().is_static() && signal_param_shape.rank().get_length() == 3 &&
+                         signal_param_shape.is_static(),
+                     "Shape of signal input must be static with the rank equal to 3.");
+
+    int64_t frame_length = signal_param_shape[axis].get_length() / frame_step;  // default value
+    if (dft_length_provided) {
+        const auto& frame_length_node = ng_inputs[3];
+        CHECK_VALID_NODE(node,
+                         ngraph::op::is_constant(frame_length_node.get_node_shared_ptr()) &&
+                             ov::shape_size(frame_length_node.get_shape()) <= 1,
+                         "frame_length input must be a scalar or Shape{1} constant.");
+        frame_length = ov::as_type_ptr<default_opset::Constant>(frame_length_node.get_node_shared_ptr())
+                           ->cast_vector<int64_t>()[0];
+    }
+
+    const auto window_node_provided = ng_inputs.size() > 2 && !ngraph::op::is_null(ng_inputs[2]);
+    if (window_node_provided) {  // window input provided
+        if (ng_inputs[2].get_partial_shape().rank().is_static()) {
+            CHECK_VALID_NODE(node,
+                             ng_inputs[2].get_partial_shape().rank().get_length() == 1,
+                             "The rank of window input must be 1D.");
+            if (ng_inputs[2].get_partial_shape()[0].is_static()) {
+                CHECK_VALID_NODE(node,
+                                 ng_inputs[2].get_partial_shape()[0].get_length() == frame_length,
+                                 "The rank of window input must be 1D.");
```

Comment:
The message about 1D rank is from the check above, something like that can be more informative:
```suggestion
                                 "The dimension of window shape must be equal to `frame_length`.");
```